redbot/core/core_commands.py

```diff
@@ -1940,6 +1941,357 @@ async def reload(self, ctx: commands.Context, *cogs: str):
             for page in pagify(total_message):
                 await ctx.send(page)
 
+    # TODO: Guild owner permissions for guild scope slash commands and syncing?
+    @commands.group()
+    @checks.is_owner()
+    async def slash(self, ctx: commands.Context):
+        """Base command for managing what application commands are able to be used on [botname]."""
+
+    @slash.command(name="enable")
+    async def slash_enable(
+        self,
+        ctx: commands.Context,
+        command_name: str,
+        command_type: Literal["slash", "message", "user"] = "slash",
+    ):
+        """Marks an application command as being enabled, allowing it to be added to the bot.
+
+        See commands available to enable with `[p]slash list`.
+
+        This command does NOT sync the enabled commands with Discord, that must be done manually with `[p]slash sync` for commands to appear in users' clients.
+
+        **Arguments:**
+            - `<command_name>` - The command name to enable. Only the top level name of a group command should be used.
+            - `[command_type]` - What type of application command to enable. Must be one of `slash`, `message`, or `user`. Defaults to `slash`.
+        """
+        command_type = command_type.lower().strip()
+
+        if command_type == "slash":
+            raw_type = discord.AppCommandType.chat_input
+        elif command_type == "message":
+            raw_type = discord.AppCommandType.message
+        elif command_type == "user":
+            raw_type = discord.AppCommandType.user
+        else:
+            await ctx.send(_("Command type must be one of `slash`, `message`, or `user`."))
+            return
+
+        current_settings = await self.bot.list_enabled_app_commands()
+        current_settings = current_settings[command_type]
+
+        if command_name in current_settings:
+            await ctx.send(_("That application command is already enabled."))
+            return
+
```

Comment:
We should validate the command being enabled against the disabled list as well here to prevent confusion when enabling individual commands. This will prevent users from being able to enable commands pre-emptively before adding a cog but it's better than them being confused why enabling a command didn't work and the list command showing extraneous unknown commands. If you want to get real fancy you can add a fuzzy search if the user makes a typo but that can be another time.